packages/nimble-components/src/table/index.ts

```diff
@@ -904,10 +923,9 @@ export class Table<
         this.tableData = rows.map(row => {
             const isGroupRow = row.getIsGrouped();
             const hasParentRow = isGroupRow ? false : row.getParentRow();
-            // we check row.original.subRows below because row.subRows is populated for group rows
-            // which we don't want to include
-            const isParent = row.original.subRows !== undefined
-                && row.original.subRows.length > 0;
+            const isParent = !isGroupRow
+                && this.isHierarchyEnabled()
```

Comment:
Should this check just move into `getRowCanExpand`?

Reply:
That's an interesting question. I actually thought about that, and I was a bit undecided myself because it isn't as straight forward as putting the check in `getRowCanExpand` because we want group rows to be expandable even when hierarchy isn't enabled. Therefore, the options I can think of are:

_Option 1:_ Keep this code as-is where the check is within `refreshRows`

_Option 2:_ Update `getRowCanExpand` to have a check that looks something like this:

```ts
private readonly getRowCanExpand = (
        row: TanStackRow<TableNode<TData>>
    ): boolean => {
        if (row.getIsGrouped()) {
            return;
        }
        return this.isHierarchyEnabled() && this.expansionManager.isRowExpandable(row);
    };
```

_Option 3:_ Update the expansion manager to know about whether or not _parentIdFieldName_ is configured

- I went with Option 1 since it was the most straight forward.
- I'm not a huge fan of Option 2 since I like the current simplicity of `getRowCanExpand` in that it is just a wrapper around the ExpansionManager
- I was concerned about the complexity of Option 3. I didn't want the ExpansionManager to have to keep track of the parentIdFieldName, but I actually think the change there could be as simple as passing the current value of parentIdFieldName to the ExpansionManager's isRowExpandable function.

I'm going to try out Option 3, and I'll go that direction unless it starts feeling overly complicated.